SpockOpaleye/src/Email.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes       #-}
+
+module Email where
+
+import           Network.Mail.Mime
+import           Network.Mail.SMTP hiding (simpleMail)
+
+import           ApiBase
+import           Conf              (apikey)
+import           Control.Lens
+import           Data.ByteString   hiding (putStrLn)
+import           Data.Monoid
+import           Data.String.Here
+import qualified           Data.Text as T
+import           DataTypes
+
+sendgridMail :: Mail -> IO ()
+sendgridMail mail = sendMailWithLogin' "smtp.sendgrid.net" 587 "apikey" apikey mail
+
+setCid :: T.Text -> T.Text -> Mail -> Mail
+setCid filename cid mail@Mail {mailParts = alternatives} = mail { mailParts = (setCidInAlternative filename cid) <$> alternatives}
+  where
+  setCidInAlternative :: T.Text -> T.Text -> Alternatives -> Alternatives
+  setCidInAlternative fp cid parts = (setCidForPart filename cid) <$> parts
+  setCidForPart :: T.Text -> T.Text -> Part -> Part
```

Comment:
Attaching a part with a given cid (or returning an auto-generated cid), IMO, should be part of the mail library's API.